includes/class-wc-gateway-stripe.php

```diff
@@ -859,7 +860,12 @@ public function process_payment( $order_id, $retry = true, $force_save_source =
 				do_action( 'wc_gateway_stripe_process_payment', $response, $order );
```

Comment:
We should be passing here the `Charge` object instead of the whole `$response` object (which can be an Intent or a Charge).

Reply:
To avoid this issue, I am extracting `$charge` and `$response_error` no matter if the charge comes from a source or an intent.

I know the _if-else_ around `if ( $prepared_source->is_intent ) {` looks a bit awkward this way, but again: In a follow-up PR we should be removing source-based charges and fully switch to PaymentIntents.  Then the condition will be redundant.